android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java

```diff
@@ -25,9 +24,24 @@
  */
 public class MusicService extends HeadlessJsTaskService {
 
+    private static final String MUSIC_SERVICE_NOTIFICATION_CHANNEL_ID = "MusicService";
```

Comment:
Is there a reason for creating a new channel? Can't you use `Utils.NOTIFICATION_CHANNEL` instead, unifying the registering code?

Reply:
Can we make sure that the `Utils.Notificaiton_CHANNEL` existed before init Music Service?

Reply:
NOTIFICATION_CHANNEL is static so should be created on class load.

Reply:
I know what you mean. You should probably move the registering code to a static method (probably in Utils), then call it whenever you need to use the channel.

Reply:
The  ExoPlayer Controls Notification is using Utils.Notificaiton_CHANNEL, I'm not if it's ok to re-create channel during Playback notification displaying?

Reply:
Yes, you can register the channel multiple times, the OS will just overwrite it when you register it again. In fact, Google recommends registering it whenever you want to display a notification, so you can be sure that the channel exists.

Reply:
Do you mean something like the following to use wherever we need a notification channel?

`Utils.getNotificationChannel`:
```java
    public static String getNotificationChannel(Context context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                Utils.NOTIFICATION_CHANNEL,
                "MusicService",
                NotificationManager.IMPORTANCE_DEFAULT
            );
            channel.setShowBadge(false);
            channel.setSound(null, null);
            ((NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE)).createNotificationChannel(channel);
        }
        return Utils.NOTIFICATION_CHANNEL;
    }
```

To use in:

https://github.com/react-native-kit/react-native-track-player/blob/6471be68dfc772bc3f94ad74d74455902651a7c8/android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java#L81

https://github.com/react-native-kit/react-native-track-player/blob/6471be68dfc772bc3f94ad74d74455902651a7c8/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java#L65

Reply:
Is there any update on this issue?

Reply:
What is needed to move forward here? Is @puckey's solution appropriate, and complete?

Reply:
> Yes, you can register the channel multiple times, the OS will just overwrite it when you register it again. In fact, Google recommends registering it whenever you want to display a notification, so you can be sure that the channel exists.

I created an other PR as your suggestion here https://github.com/react-native-kit/react-native-track-player/pull/769 
Please help to close this PR. Thank you!